tests/Functional/FunctionalTest.php

```diff
@@ -35,6 +36,7 @@ class FunctionalTest extends TestCase
 {
     private const RENDER_DOCUMENT_FILES = ['main-directive'];
     private const SKIP_INDENTER_FILES   = ['code-block-diff'];
+    private const RENDER_ALL            = ['toctree'];
```

Comment:
I'm wondering why makes these functional tests special and deserving of more test code… my understanding is that since this is about a TOC, we need many documents.  Do we really need to render them all though? Since only `index.html` is added in these commit, whereas there are many rst files, we don't actually render them all.

Reply:
> I noticed some of your commits are verified, some other are not, and I think that might be because one of the email addresses you use in your commits is not associated with your github profile. If that's on purpose you can ignore my comment, otherwise please associate your email address with your Github account, or change the email in your commits to an address already associated with it. If you do not want to expose your personal email address, you may use `$userName@users.noreply.github.com`, that way we can still reach you through Github. Additionally, you will get credit for this contribution on your Github profile.

Not sure how to do that, I always commit and push via PHP Storm. Tried to squash related commits though, maybe that somehow removed my email?

Reply:
How can you render a propper toctree without rendering all the documents? From What I understand `TocNodeRenderer` uses `ResolvedReference` which seem to equal a rendered page in some way

Reply:
Ah you're right, I didn't realize we were rendering them all, but not comparing them all with an expected rendering.

Reply:
> Not sure how to do that

Apparently your email address ends with `w-commerce.de`. Is that email address listed at https://github.com/settings/emails ?

Reply:
If we are thinking about toctrees, comparing them all is over-the top, however if we think about different directives to be tested like genindex.rst / `.. index:: ` directive we might have to test other files. So maybe make the file to be compared configurable somehow? We could of course compare the whole file tree but would then have many false negatives when something else changes

Reply:
> So maybe make the file to be compared configurable somehow?

Maybe yeah. No sure how to get this one right though (have a list of files to compare or to avoid comparing, compare only when an html/tex counterpart is available, etc.).